hexa/plugins/connector_gcs/models.py

```diff
@@ -0,0 +1,174 @@
+import typing
+from logging import getLogger
+
+from django.contrib.auth.models import AnonymousUser
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.db.models import Q
+from django.template.defaultfilters import pluralize
+from django.urls import reverse
+from django.utils.translation import gettext_lazy as _
+
+from hexa.catalog.models import Datasource
+from hexa.core.models import Base
+from hexa.core.models.base import BaseQuerySet
+from hexa.core.models.cryptography import EncryptedTextField
+from hexa.user_management import models as user_management_models
+from hexa.user_management.models import Permission, PermissionMode
+
+logger = getLogger(__name__)
+
+
+class Credentials(Base):
+    """We actually only need one set of credentials. These "principal" credentials will be then used to generate
+    short-lived credentials with a tailored policy giving access only to the buckets that the user team can
```

Comment:
I don't see the "short-lived" part - don't we use the full credentials every time at the moment?